app/src/main/java/fr/iut_rodez/pathpilot_android_client/login/LoginService.java

```diff
@@ -0,0 +1,139 @@
+package fr.iut_rodez.pathpilot_android_client.login;
+
+import static fr.iut_rodez.pathpilot_android_client.util.Network.getRequestQueue;
+import static fr.iut_rodez.pathpilot_android_client.util.VolleyErrorHandler.handleError;
+
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.util.Log;
+
+import com.android.volley.AuthFailureError;
+import com.android.volley.NetworkError;
+import com.android.volley.NoConnectionError;
+import com.android.volley.ParseError;
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.ServerError;
+import com.android.volley.TimeoutError;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.android.volley.toolbox.Volley;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Date;
+
+import fr.iut_rodez.pathpilot_android_client.BuildConfig;
+import fr.iut_rodez.pathpilot_android_client.util.Network;
+import fr.iut_rodez.pathpilot_android_client.util.Popup;
+
+public class LoginService {
+
+    private static final String LOGIN_URL = BuildConfig.API_BASE_URL + "auth/login";
+    private static RequestQueue requestQueue;
+
+    private static final String TAG = LoginService.class.getSimpleName();
+
+    /**
+     * Login the user with the given email and password.
+     *
+     * @param loginInput the email and password of the user
+     * @param context    the context of the application
+     * @throws IllegalArgumentException if email or password is empty
+     */
+    public static void login(LoginInput loginInput, Context context) {
+        Log.d(TAG, "API URL: " + LOGIN_URL);
+
+        ProgressDialog progressDialog = new ProgressDialog(context);
+        progressDialog.show();
+
+        requestQueue = getRequestQueue(context);
+        JSONObject loginInputJson = loginInput.toJson();
+
+        Log.d(TAG, "login: " + loginInputJson);
+
+        JsonObjectRequest request = new JsonObjectRequest(Request.Method.POST, LOGIN_URL, loginInputJson,
+                response -> {
+                    Log.d(TAG, "onResponse: " + response);
+                    try {
+                        String token = response.getString("token");
+                        saveAuthToken(token, context);
+                        //TODO Goto main Page
+                    } catch (JSONException e) {
+                        Log.e(TAG, "Error while parsing JSON response", e);
+                    }
+                    progressDialog.dismiss();
+                },
+                error -> {
+                    Log.e(TAG, "Error while logging in", error);
+                    progressDialog.dismiss();
+
+                    handleError(context, error);
+                }) {
+        };
+
+        requestQueue.add(request);
+    }
+
+    private static void saveAuthToken(String token, Context context) {
```

Comment:
Don't forget signature (no need for getter/setter/constructor, only for class)

Reply:
Fix in https://github.com/SAE-S5-01-Developpement-avance/PathPilot-android-client/pull/38/commits/3f381088a74825447a65980cc8bb98e00dd60954